docs/design/why_rust.md

```diff
@@ -0,0 +1,39 @@
+# Why Rust for ICU4X?
+
+The team chose Rust for writing ICU4X because it is the programming language that best fit the desired value propositions of the project.
+
+## Good for embedded systems
+
+ICU4X aims to serve the needs of embedded systems like smart watches and other IoT devices. The ecosystem surrounding Embedded Rust is strong for writing software to run on these devices.
+
+Other options for embedded systems: C, C++, MicroPython, JavaScript running on [XS](https://blog.moddable.com/blog/secureprivate/).
+
+## Good for WASM
+
+WebAssembly does not support garbage-collected languages; the WASM GC proposal aims to address this, but there is still time before it sees broad availability.
```

Comment:
suggestion: Rust has also had by far the most wasm maturity because the Rust community adopted it quick and early.

Reply:
Added, minus the superlatives.